design/plugin.md

```diff
@@ -122,6 +122,10 @@ When a cluster is attached, the CLI will also pass the following ENV variables:
                       verify server certificates against. When it is not set, the system's CA bundle
                       must be used.
 
+The CLI will also pass the subcommand config as additional environment variables. For instance, when the
+`hello` subcommand has some config values (eg. `dcos config set hello.world foo`), they will be passed when
+the subcommand is invoked (eg. `DCOS_HELLO_WORLD=foo`).
```

Comment:
I was thinking if we should make it a little more clear how `hello.world` is transformed to `DCOS_HELLO_WORLD`, i.e. it's transformed to all-uppercase, dots are replaced with underscores and the prefix `DCOS_` is prepended.

Reply:
I've updated the doc with more information.